feat(curriculum): forum leaderboard project (#50241)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-fetch-promises-and-async-await-by-building-an-fcc-forum-leaderboard/643c92bd109ab00ae3adae8d.md

---
id: 643c92bd109ab00ae3adae8d
title: Step 31
challengeType: 0
dashedName: step-31
---

# --description--

> You will add a callback function to the `forEach` method. The callback function will take the `item` as a parameter. Like this: 

Reviewer: Campers are familiar with `forEach` at this point - we don't need an example here.

> You will add a callback function to the `forEach` method. The callback function will take the `item` as a parameter. Like this: 

Author: added suggestions of both @naomi-lgbt and @zairahira

---
id: 643c92bd109ab00ae3adae8d
title: Step 14
challengeType: 0
dashedName: step-14
---

# --description--

> Add a callback function to the `forEach` method. The callback function will take the `item` as a parameter.

Reviewer 2: ```suggestion
Add a callback function to the `.forEach()` method. The callback function should take `item` as an argument.
```

# --hints--

> You should use the callback function in the `forEach` method. Don't forget to add `item` as a parameter.

Reviewer 2: ```suggestion
You should add a callback function to the `.forEach()` method that takes `item` as an argument.
```

}

.general {
  background-color: var(--light-green);
}

.javascript {
  background-color: var(--golden-yellow);
}

.backend {
  background-color: var(--gold);
}

.python {
  background-color: var(--green);
}

.motivation {
  background-color: var(--peach);
}

.avatar-container {
  display: flex;
  justify-content: center;
  gap: 10px;
  flex-wrap: wrap;
}

.avatar-container img {
  width: 30px;
  height: 30px;
}

@media (max-width: 750px) {
  .table-wrapper {
    padding: 0 15px;
  }

  table {
    width: 700px;
  }

  th {
    font-size: 1.2rem;
  }

  .post-title {
    font-size: 1.1rem;
  }
}
```

```js
const FORUM_LATEST = "https://forum-proxy.freecodecamp.rocks/latest";
const FORUM_TOPIC_URL = "https://forum.freecodecamp.org/t/";
const FORUM_CATEGORY_URL = "https://forum.freecodecamp.org/c/";
const AVATAR_URL = "https://sea1.discourse-cdn.com/freecodecamp";

const postsContainer = document.getElementById('posts-container');

const fetchData = async () => { 
  try{
    const res = await fetch(FORUM_LATEST);
    const data = await res.json();
    console.log(data)
    showLatestPosts(data);
  }
  catch(err){
    console.log(err)
  }
}

fetchData();

--fcc-editable-region--

> const showLatestPosts = (data) => {
>   const {topic_list, users} = data;
>   const {topics} = topic_list;
> 
>   topics.forEach();
> 
> }

Reviewer 2: ```suggestion
const showLatestPosts = (data) => {
  const { topic_list, users } = data;
  const { topics } = topic_list;

  topics.forEach();
};
```
